doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,2309 @@
+Ethernet0    Tx CDR LOL on host L6        False/  0/  never/  never
+Ethernet0    Tx CDR LOL on host L7        False/  0/  never/  never
+Ethernet0    Tx CDR LOL on host L8        False/  0/  never/  never
+Ethernet0    Tx EQ fault on host L1       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host L2       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host L3       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host L4       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host L5       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host L6       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host L7       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host L8       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media L1       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media L2       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media L3       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media L4       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media L5       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media L6       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media L7       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media L8       False/  0/  never/  never
+Ethernet0    Target output power out of range  False/  0/  never/  never
+Ethernet0    Fine tuning out of range flag False/  0/  never/  never
+Ethernet0    Tuning not accepted flag    False/  0/  never/  never
+Ethernet0    Invalid channel number flag False/  0/  never/  never
+Ethernet0    Tuning complete flag        False/  0/  never/  never
+```
+
+## 4. SONiC CMIS diagnostic monitoring workflow
+
+### 4.1 Static Diagnostic Information
+
+The `SfpStateUpdateTask` thread is responsible for updating the static diagnostic information for all the transceivers in the system. The static diagnostic information, such as threshold values for DOM, VDM and PM, are read from the transceiver and updated in the `redis-db` during `xcvrd` boot-up and during transceiver removal and insertion.
+
+The following tables are updated by the `SfpStateUpdateTask` thread:
+
+1. `TRANSCEIVER_DOM_THRESHOLD`
+2. `TRANSCEIVER_VDM_THRESHOLD`
+
+### 4.2 Dynamic Diagnostic Information
+
+The `DomInfoUpdateTask` thread is responsible for updating the dynamic diagnostic information for all the transceivers in the system. The `DomInfoUpdateTask` thread is triggered by a timer (`DOM_INFO_UPDATE_PERIOD_SECS`), which is set to 60 seconds by default. The `DomInfoUpdateTask` thread reads the diagnostic information from the transceiver and updates the relevant tables in `redis-db`.
+
+#### 4.2.1 High-Level Steps for Updating Dynamic Diagnostic Information
+
+1. The `DomInfoUpdateTask` thread is created by the `xcvrd` process.
+2. A timer (`DOM_INFO_UPDATE_PERIOD_SECS`) is set to read the diagnostic information from the transceiver every 60 seconds.
```

Comment:
Many new fields have been added. Is there any measurement for the time required to complete all port readings for each cycle?

Reply:
@prgeor Can you please help with this?